packages/sw-lib/src/lib/sw-lib.js

```diff
@@ -55,8 +55,37 @@ class SWLib {
   constructor({cacheId, clientsClaim, handleFetch,
                directoryIndex = 'index.html',
                precacheChannelName = 'precache-updates',
-               ignoreUrlParametersMatching = [/^utm_/],
-             } = {}) {
+               ignoreUrlParametersMatching = [/^utm_/]} = {}) {
+    if (!environment.isServiceWorkerGlobalScope()) {
+      // If we are not running in a service worker, fail early.
+      throw ErrorFactory.createError('not-in-sw');
+    }
+
+    if (environment.isDevBuild()) {
+      if (environment.isLocalhost()) {
+        // If this is a dev bundle on localhost, print a welcome message.
+        logHelper.debug({
+          message: 'Welcome to Workbox!',
+          data: {
+            '📖': 'https://googlechrome.github.io/sw-helpers/',
+            '❓': 'https://stackoverflow.com/questions/ask?tags=workbox',
+            '🐛': 'https://github.com/GoogleChrome/sw-helpers/issues/new',
+          },
+        });
+      } else {
+        // If this is a dev bundle not on localhost, recommend the prod bundle.
+        logHelper.warn(`This appears to be a production server. Please switch
+          to the smaller, optimized production build of Workbox.`);
+      }
+    } else {
+      // If this is a prod bundle on localhost, recommend the dev bundle.
+      if (environment.isLocalhost()) {
+        logHelper.warn(`This appears to be a development server. Additional
```

Comment:
I'm not crazy about spamming people with this message. There isn't anything inherently wrong with using prod on localhost.